index.bs

```diff
@@ -2042,21 +2090,21 @@ To <dfn noexport>parse source-registration JSON</dfn> given a [=byte sequence=]
 1. Let |debugReportingEnabled| be false.
 1. If |value|["`debug_reporting`"] [=map/exists=] and is a [=boolean=], set
     |debugReportingEnabled| to |value|["`debug_reporting`"].
-1. Let |eventReportWindows| be the result of [=parsing top-level report windows=] with |value|, |sourceTime|, |sourceType|, and |expiry|.
-1. If |eventReportWindows| is an error, return null.
 1. Let |aggregatableReportWindow| be a new [=report window=] with the following items:
 
     : [=report window/start=]
     :: |sourceTime|
     : [=report window/end=]
     :: |sourceTime| + |aggregatableReportWindowEnd|
 
-1. Let |triggerSpecs| be a new [=trigger spec map=].
-1. [=set/iterate|For each=] integer |triggerData| of [=the exclusive range|the range=] 0 to |triggerDataCardinality|, exclusive:
-    1. Let |spec| be a new [=trigger spec=] struct whose items are:
-        : [=trigger spec/event-level report windows=]
-        :: |eventReportWindows|
-    1. [=map/Set=] |triggerSpecs|[|triggerData|] to |spec|.
+1. Let |triggerDataMatchingMode| be "<code>[=trigger-data matching mode/modulus=]</code>".
+1. If |value|["`trigger_data_matching`"] [=map/exists=]:
+    1. If |value|["`trigger_data_matching`"] is not a [=string=], return null.
+    1. If |value|["`trigger_data_matching`"] is not a [=trigger-data matching mode=], return null.
```

Comment:
Why do we need both this check and the string check? Doesn't this one imply type matching as well?

Reply:
Done.